routing/router.go

```diff
@@ -485,11 +497,15 @@ func (r *ChannelRouter) Start() error {
 
 	// If AssumeChannelValid is present, then we won't rely on pruning
 	// channels from the graph based on their spentness, but whether they
-	// are considered zombies or not.
+	// are considered zombies or not. We will start zombie pruning after a
+	// small delay, to avoid slowing down startup of lnd.
 	if r.cfg.AssumeChannelValid {
-		if err := r.pruneZombieChans(); err != nil {
-			return err
-		}
+		time.AfterFunc(r.cfg.FirstTimePruneDelay, func() {
+			log.Info("Initial zombie prune starting")
```

Comment:
perhaps we should check the quit signal before executing this, just in case this executes around the time lnd is shutting down? some weird things will probably happen if it executes after that point.

in fact, i think it might be better to avoid using `AfterFunc`, and instead use a waitgrouped goroutine that selects on `quit` and `time.After`, just to be sure we aren't executing anything after the router has shutdown.

Reply:
Good point. I can revert it back to the original approach, where the call was simply done from the main goroutine.

Reply:
If `pruneZombieChans` checks the stopped variable, or we do it here then I don't think we have anything to worry about. Note that we moved to his approach to shrink the diff by quite a bit, and also avoid a one-time used channel that only ever triggers the select statement once in the lifetime of the daemon.

Reply:
If we shutdown _before_ this executes, afaict nothing happens in the end. 

If we shutdown _while_ this is executing then it'll pause shutdown until things are complete.

Reply:
If we're concerned with it slowing down the shutdown, then we can capture the `*time.Timer` returned here, then `defer initialPruneTimer.Stop()` which'll cancel it if it hasn't executed by the time we want to shutdown.